lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -763,155 +868,160 @@ private SyndicationContent ReadContentFrom(XmlReader reader, SyndicationItem ite
-                    if (reader.HasAttributes)
+                    while (reader.MoveToNextAttribute())
                     {
-                        while (reader.MoveToNextAttribute())
+                        if (reader.LocalName == "lang" && reader.NamespaceURI == XmlNs)
                         {
-                            if (reader.LocalName == "lang" && reader.NamespaceURI == XmlNs)
-                            {
-                                result.Language = reader.Value;
-                            }
-                            else if (reader.LocalName == "base" && reader.NamespaceURI == XmlNs)
+                            result.Language = await reader.GetValueAsync();
+                        }
+                        else if (reader.LocalName == "base" && reader.NamespaceURI == XmlNs)
+                        {
+                            result.BaseUri = FeedUtils.CombineXmlBase(result.BaseUri, await reader.GetValueAsync());
+                        }
+                        else
+                        {
+                            string ns = reader.NamespaceURI;
+                            string name = reader.LocalName;
+                            if (FeedUtils.IsXmlns(name, ns) || FeedUtils.IsXmlSchemaType(name, ns))
                             {
-                                result.BaseUri = FeedUtils.CombineXmlBase(result.BaseUri, reader.Value);
+                                continue;
                             }
-                            else
+
+                            string val = await reader.GetValueAsync();
+
+                            if (!TryParseAttribute(name, ns, val, result, this.Version))
                             {
-                                string ns = reader.NamespaceURI;
-                                string name = reader.LocalName;
-                                if (FeedUtils.IsXmlns(name, ns) || FeedUtils.IsXmlSchemaType(name, ns))
+                                if (_preserveAttributeExtensions)
                                 {
-                                    continue;
-                                }
-                                string val = reader.Value;
-                                if (!TryParseAttribute(name, ns, val, result, this.Version))
-                                {
-                                    if (_preserveAttributeExtensions)
-                                    {
-                                        result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), reader.Value);
-                                    }
-                                    else
-                                    {
-                                        TraceSyndicationElementIgnoredOnRead(reader);
-                                    }
+                                    result.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), await reader.GetValueAsync());
```

Comment:
>await reader.GetValueAsync() [](start = 128, length = 28)

Use the value stored in val. The reader hasn't been passed to any other methods since then so the value won't have changed.